dace/transformation/dataflow/tiling_with_overlap.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2019-2020 ETH Zurich and the DaCe authors. All rights reserved.
+""" This module contains classes and functions that implement the orthogonal
+    tiling with overlap transformation. """
+
+from dace import registry
+from dace.properties import make_properties, ShapeProperty
+from dace.transformation.dataflow import MapTiling
+from dace.sdfg import nodes
+from dace.symbolic import pystr_to_symbolic
+
+@registry.autoregister_params(singlestate=True)
+@make_properties
+class MapTilingWithOverlap(MapTiling):
+    """ Implements the orthogonal tiling transformation with overlap.
+
+        Orthogonal tiling is a type of nested map fission that creates tiles
+        in every dimension of the matched Map. The overlap can vary in each
+        dimension and direction. It is added to each tile and the starting
+        and end points of the outer map are ajusted to account for the overlap.
+    """
+
+    # Properties
+    lower_overlap = ShapeProperty(dtype=tuple,
+                                  default=None,
+                                  desc="Lower overlap per dimension")
+    upper_overlap = ShapeProperty(dtype=tuple,
+                                  default=None,
+                                  desc="Upper overlap per dimension")
+
+    def apply(self, sdfg):
+        if len(self.lower_overlap) == 0:
+            return
+        if len(self.upper_overlap) == 0:
+            return
+        
+        graph = sdfg.nodes()[self.state_id]
+        map_entry = graph.nodes()[self.subgraph[self._map_entry]]
+
+        # Tile the map
+        self.tile_trivial = True
+        MapTiling.apply(self, sdfg)
```

Comment:
a very strange way of invoking a `super()` method

Reply:
I replaced it.